### ISSUE TITLE
spack built pkg-config does not work properly on cray machines

### ISSUE THREAD

**Issue by krafczyk** [LLNL/spack#3167]
When I attempt to install `xproto` as part of the dependencies of `hdf5`, I get the following error:

```
spack install -j 1 -v xproto      
==> Installing xproto
==> util-macros is already installed in /mnt/a/u/sciteam/mkrafcz2/spack/spack/opt/spack/cray-CNL-interlagos/gcc-6.2.0/util-macros-1.19.1-qtnyftyaswzabuup45lt24ahxmeqsfct
==> pkg-config is already installed in /mnt/a/u/sciteam/mkrafcz2/spack/spack/opt/spack/cray-CNL-interlagos/gcc-6.2.0/pkg-config-0.29.1-6ccbb6quie45t3fia3rlhf5yuwl35tfb
==> Using cached archive: /mnt/a/u/sciteam/mkrafcz2/spack/spack/var/spack/cache/xproto/xproto-7.0.29.tar.gz
==> Already staged xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p in /mnt/a/u/sciteam/mkrafcz2/spack/spack/var/spack/stage/xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p
==> Already patched xproto
==> Building xproto [AutotoolsPackage]
==> Executing phase : 'autoreconf'
==> Executing phase : 'configure'
==> '/mnt/a/u/sciteam/mkrafcz2/spack/spack/var/spack/stage/xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p/xproto-7.0.29/configure' '--prefix=/mnt/a/u/sciteam/mkrafcz2/spack/spack/opt/spack/cray-CNL-interlagos/gcc-6.2.0/xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p'
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... /mnt/a/u/sciteam/mkrafcz2/spack/spack/lib/spack/env/gcc/gcc
checking whether the C compiler works... no
configure: error: in `/tmp/mkrafcz2/spack-stage/spack-stage-FMhtvi/xproto-7.0.29':
configure: error: C compiler cannot create executables
See `config.log' for more details
==> Error: ProcessError: Command exited with status 77:
    '/mnt/a/u/sciteam/mkrafcz2/spack/spack/var/spack/stage/xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p/xproto-7.0.29/configure' '--prefix=/mnt/a/u/sciteam/mkrafcz2/spack/spack/opt/spack/cray-CNL-interlagos/gcc-6.2.0/xproto-7.0.29-3ha6snz2ux67bbtgn53g4utxxawywo2p'
/mnt/a/u/sciteam/mkrafcz2/spack/spack/lib/spack/spack/build_systems/autotools.py:268, in configure:
     261      def configure(self, spec, prefix):
     262          """Runs configure with the arguments specified in :py:meth:`.configure_args`
     263          and an appropriately set prefix.
     264          """
     265          options = ['--prefix={0}'.format(prefix)] + self.configure_args()
     266  
     267          with working_dir(self.build_directory, create=True):
  >> 268              inspect.getmodule(self).configure(*options)

See build log for details:
  /tmp/mkrafcz2/spack-stage/spack-stage-FMhtvi/xproto-7.0.29/spack-build.out
```

Looking at `config.log` I have series of lines like this:

```
configure:3344: checking for C compiler version
configure:3353: /mnt/a/u/sciteam/mkrafcz2/spack/spack/lib/spack/env/gcc/gcc --version >&5
Error invoking pkg-config!
Package lsb-cray-hss was not found in the pkg-config search path.
Perhaps you should add the directory containing `lsb-cray-hss.pc'
to the PKG_CONFIG_PATH environment variable
Package 'lsb-cray-hss', required by 'cray-rca', not found
```

Attempting to run the gcc wrapper with `spack env xproto` gives the following:

```
spack env xproto /mnt/a/u/sciteam/mkrafcz2/spack/spack/lib/spack/env/gcc/gcc --version
Error invoking pkg-config!
Package lsb-cray-hss was not found in the pkg-config search path.
Perhaps you should add the directory containing `lsb-cray-hss.pc'
to the PKG_CONFIG_PATH environment variable
Package 'lsb-cray-hss', required by 'cray-rca', not found
```

I've determined that this happens for any package which has `pkg-config` as a dependency.

It seems that spack built `pkg-config` has some kind of problem with the cray environment. `spack env bzip2` is just fine:

```
spack env bzip2 /mnt/a/u/sciteam/mkrafcz2/spack/spack/lib/spack/env/gcc/gcc --version 
gcc (GCC) 6.2.0 20160822 (Cray Inc.)
Copyright (C) 2016 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

Any ideas why this might be happening?

**Comment by krafczyk** [comment]
Okay, with that corroboration, I've ammended the title.

**Comment by krafczyk** [comment]
Okay, small update, it seems that the cray `pkg-config` is built with different options than spack's because inside the executable there are additional `PKG_CONFIG_PATH`s which are searched. A couple are: `/usr/lib64/pkgconfig`, `/usr/local/lib64/pkgconfig`, stuff of that nature. This is where all the missing `.pc` files are located.

**Comment by adamjstewart** [comment]
Closed by #3378? Reopen if not.
